BaseClasses.py

```diff
@@ -831,6 +835,24 @@ def get_connecting_entrance(self, is_main_entrance: typing.Callable[[Entrance],
         for entrance in self.entrances:  # BFS might be better here, trying DFS for now.
             return entrance.parent_region.get_connecting_entrance(is_main_entrance)
 
+    def add_locations(self, locations: Dict[str, Optional[str]], location_type: Optional[typing.Type[Location]] = None) -> None:
+        """Adds locations to the Region object, where LocationType is your Location class and locations is a dict of
+        location names to address."""
+        if location_type is None:
+            location_type = Location
+        for location, address in locations.items():
+            self.locations.append(location_type(self.player, location, address, self))
+
+    def add_exits(self, exits: Dict[str, List[str, Optional[Callable]]]) -> None:
```

Comment:
i thought about a tuple as well and that didn't work too well. having Union would be functional but i just don't think it works well. i haven't thought of a good solution for this specifically that would allow you to add rules dynamically during creation. tempted to just drop the rule addition entirely and just make it a `Union[List[str], Dict[str, str]]`

Reply:
Could make it two dicts, one for the connections and one for rules.